schemas/user.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require("mongoose");
+
+const userSchema = mongoose.Schema({
+  _id: mongoose.Schema.Types.ObjectId,
+  firstName: String,
+  lastName: String,
+  telephone: String,
+  nickName: String,
+  location: String,
+  password: String,
+  email: String,
+  favoriteProducts: [
```

Comment:
Probably `favoriteProducts` and `viewedProducts` would be better to be an array of ObjectID ids. Then it would be a lot more easier to fetch them with user by one DB query